src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs

```diff
@@ -53,6 +54,24 @@ public async Task<GetResult> GetDocumentById(IGetRequest getRequest)
         }
     }
 
+    public async Task<GetResult> GetDocumentByResourceName(IGetRequest getRequest, int offset, int limit)
```

Comment:
As an FYI, everything needed to represent a request to a backend action belongs in that backend action's request object. In this case you are actually implementing a slice of `Query` so you already have `IQueryRequest` available to you with `offset` and `limit` in it.

Reply:
Really I should point out that you are actually implementing `public async Task<QueryResult> QueryDocuments(IQueryRequest queryRequest)` down at the bottom of the file.